test/test_constraints_file.py

```diff
@@ -50,18 +50,111 @@ def test_add_at_least_one_employee_per_shift_constraint():
     model = cp_model.CpModel()
 
     shifts = generate_shift_employee_combinations(employees, [test_shift], model)
-    add_at_least_one_employee_per_shift_constraint([test_shift], employees, model, shifts)
+    add_exactly_one_employee_per_shift_constraint([test_shift], employees, model, shifts)
 
     solver = cp_model.CpSolver()
     status = solver.Solve(model)
 
     assert (status != cp_model.OPTIMAL)
+
+
+# A Test to check if there is an optimal solution with 2 employees and 3 shifts.
+# Expected result is an Optimal solution because there is a solution where no employee is assigned to shifts.
+def test_add_at_most_one_shift_per_employee_in_the_same_day_constraint_without_add_exactly_one_employee_per_shift_constraint():
+    test_employee = Employee("test", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+    test_employee2 = Employee("test2", EmployeePriorityEnum.HIGHEST, EmployeeStatusEnum.senior_employee, employee_id=uuid4())
+    test_shift1 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.MORNING, start_time=datetime.datetime(2023, 12, 11, 9, 30), end_time=datetime.datetime(2023, 12, 11, 16, 0))
+    test_shift2 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.EVENING, start_time=datetime.datetime(2023, 12, 11, 16, 0), end_time=datetime.datetime(2023, 12, 11, 22, 0))
+    test_shift3 = Shift(shift_id=uuid4(), shift_type=ShiftTypesEnum.CLOSING, start_time=datetime.datetime(2023, 12, 11, 17, 30), end_time=datetime.datetime(2023, 12, 12, 2, 0))
+
+    shifts = [test_shift1, test_shift2, test_shift3]
+    employees = [test_employee, test_employee2]
+    model = cp_model.CpModel()
+    all_shifts = generate_shift_employee_combinations(employees, shifts, model)
+
+    add_at_most_one_shift_per_employee_in_the_same_day_constraint(shifts, employees, model, all_shifts)
+
+    solver = cp_model.CpSolver()
+    status = solver.Solve(model)
+
+    # there is a solution where employees are not assigned to shifts, which is considered an optimal solution.
+    assert (status == cp_model.OPTIMAL)
+    expected_employee_working = False
+
+    for shift in shifts:
+        for employee in employees:
+            key = FrozenShiftCombinationsKey(employee.employee_id, shift.shift_id)
+            working_assignment = all_shifts[key]
+            # the employee does not work this shift
+            assert (solver.Value(working_assignment) == expected_employee_working)
+
+
+# A Test to check if there is an optimal solution with 2 employees and 3 shifts. With at least one employee that is
+# working each shift. There is no optimal solution because there are fewer employees than the shift in the same day.
+def test_add_at_most_one_shift_per_employee_in_the_same_day_constraint_with_at_least_one_employee_in_a_shift_with_3_shifts_2_employees():
```

Comment:
the number doesnt matter. what is the actual case you are testing. i dont care about the constraint!!

`test_two_employees_cannot_work_the_same_shift`